packages/neo-one-node-vm/src/__tests__/syscalls/binary.test.ts

```diff
@@ -0,0 +1,207 @@
+              .writeVarBytesLE(utils.toSignedBuffer(new BN('10000000000000', 10)))
+              .toBuffer(),
+          )
+          .toBuffer(),
+      ),
+    ],
+
+    args: [[new BN('10000000000000', 10)]],
+    gas: FEES[100_000],
+  },
+
+  {
+    name: 'System.Binary.Serialize',
+    result: [],
+    error: 'Item too large',
+
+    args: [Buffer.alloc(1024 * 1024)],
+    gas: FEES[100_000],
+  },
+
+  {
+    name: 'System.Binary.Serialize',
+    result: [
+      new BufferStackItem(
+        new BinaryWriter()
+          .writeUInt8(StackItemType.Map)
+          .writeVarUIntLE(1)
+          .writeBytes(
+            new BinaryWriter()
+              .writeUInt8(StackItemType.ByteArray)
+              .writeVarBytesLE(Buffer.from('key', 'utf8'))
+              .toBuffer(),
+          )
+          .writeBytes(
+            new BinaryWriter()
+              .writeUInt8(StackItemType.ByteArray)
+              .writeVarBytesLE(Buffer.from('value', 'utf8'))
+              .toBuffer(),
+          )
+          .toBuffer(),
+      ),
+    ],
+
+    args: [
+      {
+        type: 'calls',
+        calls: [
+          {
+            name: 'NEWMAP',
+            type: 'op',
+          },
+
+          {
+            name: 'DUP',
+            type: 'op',
+          },
+
+          {
+            name: 'SETITEM',
+            type: 'op',
+            args: [Buffer.from('value', 'utf8'), Buffer.from('key', 'utf8')],
+          },
+        ],
+      },
+    ],
+
+    gas: FEES[100_000],
+  },
+
+  {
+    name: 'System.Binary.Deserialize',
+    result: [new BufferStackItem(Buffer.alloc(10, 1))],
+    args: [
+      new BinaryWriter()
+        .writeUInt8(StackItemType.ByteArray)
+        .writeVarBytesLE(Buffer.alloc(10, 1))
+        .toBuffer(),
+    ],
+
+    gas: FEES[500_000],
+  },
+
+  {
+    name: 'System.Binary.Deserialize',
+    result: [
+      new MapStackItem({
+        referenceKeys: new Map([
+          [
+            new BufferStackItem(Buffer.from('key', 'utf8')).toStructuralKey(),
+            new BufferStackItem(Buffer.from('key', 'utf8')),
+          ] as const,
+        ]),
+        referenceValues: new Map([
+          [
+            new BufferStackItem(Buffer.from('key', 'utf8')).toStructuralKey(),
+            new BufferStackItem(Buffer.from('value', 'utf8')),
+          ] as const,
+        ]),
+      }),
+    ],
+
+    args: [
+      new BinaryWriter()
+        .writeUInt8(StackItemType.Map)
+        .writeVarUIntLE(1)
+        .writeBytes(
+          new BinaryWriter()
+            .writeUInt8(StackItemType.ByteArray)
+            .writeVarBytesLE(Buffer.from('key', 'utf8'))
+            .toBuffer(),
+        )
+        .writeBytes(
+          new BinaryWriter()
+            .writeUInt8(StackItemType.ByteArray)
+            .writeVarBytesLE(Buffer.from('value', 'utf8'))
+            .toBuffer(),
+        )
+        .toBuffer(),
+    ],
+
+    gas: FEES[500_000],
+  },
+] as readonly TestCase[];
```

Comment:
Just noticed that we've been type casting this array, which seems less safe than tagging the array as `SYSCALLS: readonly TestCase[]`